app/src/test/java/com/jalasoft/routesapp/util/algorithm/RouteCalculatorTest.kt

```diff
@@ -0,0 +1,105 @@
+    val hiltRule = HiltAndroidRule(this)
+
+    private lateinit var lines: List<LinePath>
+
+    @Before
+    public override fun setUp() {
+        super.setUp()
+        val start1 = RouteAlgorithmFakeData.coordinatesToLocation(-16.52035351419114, -68.12580890707301)
+        val end1 = RouteAlgorithmFakeData.coordinatesToLocation(-16.524285569842718, -68.12298370418992)
+        val points1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points1Array)
+        val stops1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops1Array)
+        val line1 = LinePath("Line 1", "Bus", points1, start1, end1, stops1)
+
+        val start2 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5255314, -68.1254204)
+        val end2 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5241937, -68.1204527)
+        val points2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points2Array)
+        val stops2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops2Array)
+        val line2 = LinePath("246", "Mini", points2, start2, end2, stops2)
+
+        val start3 = RouteAlgorithmFakeData.coordinatesToLocation(-16.52447, -68.12538)
+        val end3 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5244779, -68.1253892)
+        val points3 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points3Array)
+        val stops3 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops3Array)
+        val line3 = LinePath("LA", "Metro", points3, start3, end3, stops3)
+        hiltRule.inject()
+        lines = listOf(line1, line2, line3)
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near the given points then it returns a list of single lines`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52094, -68.12419)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52355, -68.12269)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        assertTrue(result.isNotEmpty())
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near two lines then it returns a list of lines with two transport`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52094, -68.12419)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52442, -68.12036)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        if (result.isNotEmpty()) {
+            assertNotNull(result.first().connectionPoint)
```

Comment:
Almost there, not only assert not null, assert that the value is what you expect it to be

Reply:
Added asserts for expected results